### ISSUE TITLE
Сделать так, чтобы видимая область следовала курсору проигрывания

### ISSUE THREAD

**Issue by artyompal** [doomsower/melody#49]
...как в музыкальных редакторах. Например, если курсор выходит за пределы, то ставить левый край окна по курсору.

**Comment by doomsower** [comment]
Тут также возникает вопрос про браш. Сейчас там есть курсор, но клик по области браша его не двигает, а двигает выделение. Таким образом, курсор можно все равно вывести за видимую область. Надо ли тут что-то менять?

**Comment by artyompal** [comment]
В состоянии Paused да, курсор можно вывести за видимую область.
В состоянии Playing нельзя. Т.е. либо запретить вообще, либо периодически проверять, что курсор находится в видимой области, и если нет, то ставить левый край окна по курсору.

**Comment by artyompal** [comment]
Вроде так во всех аудио- и видеоредакторах делается.

**Comment by doomsower** [comment]
> В состоянии Paused да, курсор можно вывести за видимую область.
И что, после нажатия Play будет лютый скачок выделения к курсору?

**Comment by artyompal** [comment]
Да. Но это же один раз. Имхо так лучше, чем когда он где-то за пределами видимости играет.

**Comment by doomsower** [comment]
Я вот просто думаю, как это согласуется с перемещением выделения. Т.е. если клик переносит курсор, то выдление можно переместить только путем перетаскивания. По-моему это не очень удобно. Сейчас там click-to-recenter (правда оно сломалось, на это есть баг).

**Comment by artyompal** [comment]
В паузе клик переносит курсор, а в playing курсор сам двигается.

**Comment by artyompal** [comment]
Если сделать более удобные средства прокрутки по горизонтали, то можно этого не делать.
Например, можно сделать кнопки "экран влево", "экран вправо".

**Comment by artyompal** [comment]
В идеале, надо сделать галочку "Follow cursor", и если она установлена, то имеет место поведение, описанное мной в постах 1 и 3.

**Comment by doomsower** [comment]
У нас есть сейчас тычка для проигрывания только выделенной области. Она не очень совместима с галочкой "Follow Cursor". Как будем их совмещать?
